sdk/Trace/SpanOptions.php

```diff
@@ -35,7 +37,14 @@ public function setSpanName(string $name): API\SpanOptions
 
     public function setSpanKind(int $spanKind): API\SpanOptions
     {
-        // TODO: Implement setSpanKind() method.
+        $reflection = new ReflectionClass(API\SpanKind::class);
```

Comment:
Taking into account that an application may create a lot of spans, this cand have an impact on performance. Usually, the reflection has an impact on performance. I made [a test](https://pastebin.com/FXBN1Baf) comparing the reflection method vs a simple array and the impact in performance is around 2.5x - 3x.

That being said, I recommend replacing the reflection with a simple array(a list of spans kind) and the current kind to be checked on that.

Reply:
Yes, I thought about that but considering that right now there is no const TYPES with an array of available types, I finally decided against messing with the SpanKind interface and save the flexibility by getting available kinds via reflection dynamically.

Do you think it's okay to add const TYPES in the interface? The alternative is to create here a static getter that would get available types via reflection only once? Something like
```
protected static $types = [];

protected static function getAvailableTypes(): array
{
    if (empy(static::$types)) {
        $reflection = new ReflectionClass(API\SpanKind::class);
        static::$types = $reflection->getConstants();
    }

    return static::$types;
}
```
and so the validation would look like `if (!in_array($spanKind, static::getAvailableTypes(), true))`
The advantage of this is that we don't need to care about TYPES consistency and the perfomance impact is much less.

Reply:
All of these type stuff is implemented really good inside `MyCLabs\Enum`. Maybe we should consider switching to it instead? The why's are covered here:  
https://github.com/myclabs/php-enum#why

Reply:
I like the ideea of having an array of Types on SpanKind API.

@wRLSS the 'php-enum' uses Reflection behind the scene, so is not an improvement on the current implementation.

Reply:
There are a lot more to it. First of all it has internal cache, also it introduces typed value object which is much safer to pass around.